sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,2630 @@
+        }
+        return response;
+      }
+
+      static BlobContentInfo CommitBlockList(
+          Azure::Core::Context context,
+          Azure::Core::Http::HttpPipeline& pipeline,
+          const std::string& url,
+          const CommitBlockListOptions& options)
+      {
+        auto request = CommitBlockListConstructRequest(url, options);
+        auto response = pipeline.Send(context, request);
+        return CommitBlockListParseResponse(*response);
+      }
+
+      struct GetBlockListOptions
+      {
+        std::string Version;
+        std::string Date;
+        BlockListTypeOption ListType = BlockListTypeOption::All;
+      }; // struct GetBlockListOptions
+
+      static Azure::Core::Http::Request GetBlockListConstructRequest(
+          const std::string& url,
+          const GetBlockListOptions& options)
+      {
+        auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, url);
+        request.AddHeader("Content-Length", "0");
+        request.AddQueryParameter("comp", "blocklist");
+        std::string block_list_type_option = BlockListTypeOptionToString(options.ListType);
+        if (!block_list_type_option.empty())
+        {
+          request.AddQueryParameter("blocklisttype", block_list_type_option);
+        }
+        request.AddHeader("x-ms-version", options.Version);
+        request.AddHeader("x-ms-date", options.Date);
+        return request;
+      }
+
+      static BlobBlockListInfo GetBlockListParseResponse(Azure::Core::Http::Response& http_response)
+      {
+        BlobBlockListInfo response;
+        auto http_status_code
+            = static_cast<std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
+                http_response.GetStatusCode());
+        if (!(http_status_code == 200))
+        {
+          throw std::runtime_error("HTTP status code " + std::to_string(http_status_code));
+        }
+        response.Version = http_response.GetHeaders().at("x-ms-version");
+        response.Date = http_response.GetHeaders().at("Date");
+        response.RequestId = http_response.GetHeaders().at("x-ms-request-id");
+        auto response_clientrequestid_iterator
+            = http_response.GetHeaders().find("x-ms-client-request-id");
+        if (response_clientrequestid_iterator != http_response.GetHeaders().end())
+        {
+          response.ClientRequestId = response_clientrequestid_iterator->second;
+        }
+        response.ETag = http_response.GetHeaders().at("ETag");
+        response.LastModified = http_response.GetHeaders().at("Last-Modified");
+        response.ContentType = http_response.GetHeaders().at("Content-Type");
+        response.ContentLength
+            = std::stoull(http_response.GetHeaders().at("x-ms-blob-content-length"));
+        // TODO: Think about how to initialize
+        // xmlInitParser();
+        // TODO: Think about how to free doc on exception
+
+        // TODO: Think about how to hanlde xml > 2GB
```

Comment:
>Think about how to hanlde xml > 2GB [](start = 17, length = 35)

XML should not be larger than 2GB.
max result is 50000 with no more than 2K per item, which is 100MB.

Reply:
i'm not sure, but 2GB should be enough for now. Let's come back to this later